src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -3465,30 +3468,28 @@ void GetInputZoneEvaporativeCoolerUnit(EnergyPlusData &state)
-                Fans::GetFanIndex(state, thisZoneEvapUnit.FanName, thisZoneEvapUnit.FanIndex, errFlag, CurrentModuleObject);
-                thisZoneEvapUnit.FanInletNodeNum =
-                    Fans::GetFanInletNode(state, thisZoneEvapUnit.FanObjectClassName, thisZoneEvapUnit.FanName, errFlag);
-                thisZoneEvapUnit.FanOutletNodeNum =
-                    Fans::GetFanOutletNode(state, thisZoneEvapUnit.FanObjectClassName, thisZoneEvapUnit.FanName, errFlag);
-                Fans::GetFanVolFlow(state, thisZoneEvapUnit.FanIndex, FanVolFlow);
-                thisZoneEvapUnit.ActualFanVolFlowRate = FanVolFlow;
-                // Get the fan's availability schedule
-                thisZoneEvapUnit.FanAvailSchedPtr =
-                    Fans::GetFanAvailSchPtr(state, thisZoneEvapUnit.FanObjectClassName, thisZoneEvapUnit.FanName, errFlag);
-                if (errFlag) {
-                    ShowContinueError(state, format("...specified in {} = {}", CurrentModuleObject, thisZoneEvapUnit.Name));
+
+            thisZoneEvapUnit.fanType = static_cast<DataHVACGlobals::FanType>(getEnumValue(DataHVACGlobals::fanTypeNamesUC, Alphas(7)));
+            if (thisZoneEvapUnit.fanType == DataHVACGlobals::FanType::Invalid) {
+                ShowSevereInvalidKey(state, eoh, cAlphaFields(7), Alphas(7));
+                ErrorsFound = true;
+            } else if (thisZoneEvapUnit.fanType != DataHVACGlobals::FanType::SystemModel) {
+                thisZoneEvapUnit.FanIndex = Fans::GetFanIndex(state, thisZoneEvapUnit.FanName);
+                if (thisZoneEvapUnit.FanIndex == 0) {
+                    ShowSevereItemNotFound(state, eoh, cAlphaFields(8), thisZoneEvapUnit.FanName);
                     ErrorsFound = true;
+                } else {
+                    assert(thisZoneEvapUnit.fanType == Fans::GetFanType(state, thisZoneEvapUnit.FanIndex));
+                    thisZoneEvapUnit.FanInletNodeNum = Fans::GetFanInletNode(state, thisZoneEvapUnit.FanIndex);
+                    thisZoneEvapUnit.FanOutletNodeNum = Fans::GetFanOutletNode(state, thisZoneEvapUnit.FanIndex);
+                    thisZoneEvapUnit.ActualFanVolFlowRate = Fans::GetFanVolFlow(state, thisZoneEvapUnit.FanIndex);
+                    thisZoneEvapUnit.FanAvailSchedPtr = Fans::GetFanAvailSchPtr(state, thisZoneEvapUnit.FanIndex);
                 }
-            } else if (Util::SameString(thisZoneEvapUnit.FanObjectClassName, "Fan:SystemModel")) {
-
-                thisZoneEvapUnit.FanType_Num = DataHVACGlobals::FanType_SystemModelObject;
+            } else {
                 state.dataHVACFan->fanObjs.emplace_back(new HVACFan::FanSystem(state, thisZoneEvapUnit.FanName)); // call constructor
-                thisZoneEvapUnit.FanIndex = HVACFan::getFanObjectVectorIndex(state, thisZoneEvapUnit.FanName);
+                thisZoneEvapUnit.FanIndex = state.dataHVACFan->fanObjs.size() - 1;
```

Comment:
No need to search for this object, since you know it's going to be at the end (you just put it there).